content/blog/2024/10/what-is-the-esp-registry/index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "What is the ESP-Registry?"
+date: 2024-10-27
+showAuthor: false
+authors:
+  - "pedro-minatel"
+tags: ["Registry", "Component", "ESP-IDF"]
+---
+
+When starting a new project, we often reuse code from existing components, such as libraries and drivers, to save time. This not only saves time but also makes your project easier to maintain.
+
+In this article, we will explore what the ESP-Registry is and the key concepts of components, making it easy to understand not only how to use components but also how to contribute.
+
+## Concept
+
+In the [ESP-IDF](https://idf.espressif.com), components are widely used to make the framework more modular, making it clearer to understand and maintain. The naming convention in the ESP-IDF is to use the term "component" instead of "library" because the ESP-IDF is more like an operating system with self-contained modules that interact with the system at a higher level.

```

Comment:
```suggestion
The architecture of [ESP-IDF](https://idf.espressif.com) is modular, making it easier to understand and maintain. Such modules are called *components*, as opposed to *libraries*, because they are self-contained and used on top of the ESP-IDF core.
```
Suggest not using the term operating system, because it implies that the system is run on top of the hardware and interacts with it. ESP-IDF is not run on hardware, it is a framework that is used to generate applications which are run on hardware.

Reply:
Using the word "operating system like" is common for the ESP-IDF. But we can avoid it here and open another discussion about it.